File: python/measurement.cpp
/*
ID: happyn61
TASK: measurement
LANG: C++
*/
#include <iostream>
#include <fstream>
#include <string>

using namespace std;

int main() {
    ofstream fout ("measurement.out");
    ifstream fin ("measurement.in");
    fin >> N;

    int a=0,b=0,c=0;

    fin >> bb >> b;
    fin >> cc >> c;
    for(int i=0;i<100;i++){
        if (i%3==0)
        {
            if (b+a<=bb){
                 b=b+a;
                 a=0;
             }
             else{
                 a=a+b-bb;
                 b=bb;
             }
        }
        if (i%3==1)
        {
            if (c+b<=cc){
                 c=b+c;
                 b=0;
             }
             else{
                 b=c+b-cc;
                 c=cc;
             }
        }
        if (i%3==2)
        {
            if (c+a<=aa){
                 a=c+a;
                 c=0;
             }
             else{
                 c=a+c-aa;
                 a=aa;
             }
        }
        cout << a <<" "<<b<<" " <<c<<endl;
    }
    fout << a <<endl<<b<<endl <<c<<endl;
    return 0;
}

File: python/measurement.py
"""
ID: happyn61
LANG: PYTHON3
PROB: measurement
"""
import heapq
fin = open ('measurement.in', 'r')
fout = open ('measurement.out', 'w')
N,G=list(map(int,fin.readline().strip().split()))
ll=[]
d={} #index to amount
dd={} #amount to set
k=0
ds={}
ans=0
#from datetime import datetime
#print(datetime.now())
for i in range(N):
    a,b,c=list(map(int,fin.readline().strip().split()))
    ll.append((a,b,c))
    #print(a,b,c)
ll.sort()
m=0
h=[]
oldset=set()
for a,b,c in ll:
    if b not in d:
        d[b]=c
        if d[b] in dd:
            dd[d[b]].add(b)
        else:
            dd[d[b]]=set()
            dd[d[b]].add(b)
        heapq.heappush(h,-d[b])    
        
    else:
        #print(d,dd,b)
        dd[d[b]].remove(b)
        d[b]+=c
        if d[b] in dd:
            dd[d[b]].add(b)
        else:
            dd[d[b]]=set()
            dd[d[b]].add(b)
        heapq.heappush(h,-d[b])
    while heapq.nsmallest(1,h)==set():
        heapq.heappop(h)
    #print("==",b,c,d,dd,h)
    if len(h)==0:
        if oldset!=set():
            ans+=1
            oldset=set()
    else:
        
        if oldset!=dd[-h[0]]:
            ans+=1
            #print(ans,oldset,dd[-h[0]])
            oldset=dd[-h[0]].copy()
#print(datetime.now())
print(ans)
fout.write(str(ans)+"\n")
                
fout.close()
    
